### PR TITLE
270 Missing fields when drafting DataCite DOI

### PR DESCRIPTION
**Pull request by sorochak**
This PR solves the issue: https://github.com/cioos-siooc/metadata-entry-form/issues/270

* added version to datacite json if it exists in the metadata record
* added fundingReferences array for any organizations selected as 'Funder'
* fixed test files so the `recordToDataCite` test passes

### CODE REVIEW

**Review by fostermh — APPROVED** [review]
this appears to fix the 3 tasks in the issue

**Comment by timvdstap** [comment]
Using the preview function I can't seem to generate or update a DOI?

**Comment by timvdstap** [comment]
Hey @sorochak, the version element maps over nicely! 👍 The funder also maps over into 'Funder Name', which is great. However, the 'Funder Identifier' (https://doi.org/10.12039/100014446) and 'Funder Identifier Type (Crossref Funder ID) don't map I'm afraid. 

image

**Comment by sorochak** [comment]
Hey @timvdstap, in issue https://github.com/cioos-siooc/metadata-entry-form/issues/270, funderName was the only field requested, so that's what was added. 

We can certainly add 'Funder Identifier' as that is a field in the metadata entry form. However, I'm not sure if Funder Identifier Type' is something that we record in the metadata entry form? Can you confirm whether Identifier type is recorded? It looks like we use ORCID identifiers. So maybe Identifier Yype can be hardcoded to ORCID or 'ROR' if an ORCID identifier exists? 

Screen Shot 2024-01-15 at 2 49 46 PM

**Comment by timvdstap** [comment]
Haha woops - can't really expect you to add a feature if we didn't ask for it to be included, my bad!

I'll loop in @JessyBarrette and @Br-Johnson for their input. For the majority, if not all of the Hakai records, Tula Foundation would be the funder, and our approach would be to add Tula Foundation by searching for the it through the ROR search tool.  Not sure if Identifier Type should be hardcoded? Would we ever anticipate a funder to not be an organization?

**Comment by sorochak** [comment]
@timvdstap I guess I meant that if an ORCID exists in the record, we can set `Identifier Type` to ROR? I'm not super familiar with these types of identifiers.

**Comment by fostermh** [comment]
have a look at https://support.datacite.org/docs/schema-optional-properties-v43#19-fundingreference

Screenshot 2024-01-16 at 9 51 31 AM

**Comment by sorochak** [comment]
@fostermh right, but we only reference ROR in the metadata form?

**Comment by JessyBarrette** [comment]
I'm not 100% sure of the present discussion but in my mind, the funder should be map to which ever contact that has the role `Funder` associated to it. 

We may likely wanna map only to the ORCID id if available otherwise the ROR. But I would think these role should rarely be attached a specific person and more an institution.

**Comment by Br-Johnson** [comment]
> @timvdstap I guess I meant that if an ORCID exists in the record, we can set `Identifier Type` to ROR? I'm not super familiar with these types of identifiers.

We can only have organizations listed as Funder. If you look at the `funderIdentifierType` list that Matt posted from DataCite, it doesn't include ORCIDs. We should hardcode `funderIdentifierType` to ROR.

That being said there's currently no way in the form to prevent/restrict a person from being listed as a funder. Once solution would be grey out the input fields related to individuals if Funder is checked off?

Rather than CrossRef Funder IDs, we should just stick with using RORs for the  `funderIdentifier` and ROR for the `funderIdentifierType` since the form already supports that and more importantly because the [Open Funder Registry is deprecating and being replaced by ROR.](https://www.crossref.org/blog/open-funder-registry-to-transition-into-research-organization-registry-ror/)

Tim is right that basically all Hakai products should have Tula Foundation listed as the Funder. 

So in summary: 

IF Funder is checked in the metadata intake form
THEN an ROR is required to be entered by the user
AND Individual details (ORCID, Given name(s), Last name, position, email) are not allowed,
AND DataCite JSON contains:
```
 "fundingReferences": [
    {
      "awardUri": null,
      "awardTitle": null,
      "funderName": "Tula Foundation",
      "awardNumber": null,
      "funderIdentifier": "https://ror.org/0029jxk29",
      "funderIdentifierType": "ROR"
    }
  ]
```

**Comment by Br-Johnson** [comment]
Would you guys support having Funder as a required field in the Hakai Catalogue? I'm currently interrogating the DataCite API daily to check all our records have this and then go back and add Funder if I'm notified it's missing for a specific record. It would be great if that was obviated by the form.

The advantage to having Funder as a required field is that it links our scholarly outputs to the Tula Foundation in addition to the Hakai Institute in the PID Graph, and it's also a requirement of our Open Science Policy to acknowledge Tula Foundation funding in research outputs.

I can make a separate issue for this if there aren't any objections..

**Comment by Br-Johnson** [comment]
I just chatted with Jessy and Tim about this and came to consensus we don't need to make having a funder a requirement. Also, no need to grey out individual level details if Funder is selected. We will just add a checkllist item to our metadata review template to make sure Hakai records have Tula listed as a funder.

So, just need to make sure the ROR URL goes into the funderIdentifier field if one is entered, and "ROR" is encoded in the funderIdentifierType if an ROR URL is entered.

**Comment by sorochak** [comment]
updated Datacite JSON mapping to add `funderIdentifier` and `"funderIdentifierType": "ROR"` if provided.

example of JSON output of a test record I created:

"fundingReferences": [
        {
          "funderName": "Hakai Institute"
        },
        {
          "funderName": "Royal Roads University",
          "funderIdentifier": "https://ror.org/05w4ste42",
          "funderIdentifierType": "ROR"
        }
      ],

**Comment by timvdstap** [comment]
This seems to work great now! :-)

**Review by timvdstap — APPROVED** [review]
